internal/server/auth/http.go

```diff
@@ -0,0 +1,42 @@
+package auth
+
+import (
+	"net/http"
+)
+
+var (
+	stateCookieKey = "flipt_client_state"
+)
+
+// Middleware contains various extensions for appropriate integration of the generic auth services
+// behind gRPC gateway. This currently includes clearing the appropriate cookies on logout.
+type Middleware struct{}
+
+// NewHTTPMiddleware constructs a new auth HTTP middleware.
+func NewHTTPMiddleware() *Middleware {
+	return &Middleware{}
+}
+
+// Handler is a http middleware used to decorate the auth provider gateway handler.
+// This is used to clear the appropriate cookies on logout.
+func (m Middleware) Handler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPut || r.URL.Path == "/auth/v1/self/expire" {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		for _, cookieName := range []string{stateCookieKey, tokenCookieKey} {
+			cookie := &http.Cookie{
+				Name:   cookieName,
```

Comment:
Unsure if necessary or not. But should we config this with the same `Domain` from config?

Trying to think of where they might mismatch, and can't think right now where that would be possible. However, if the requested domain doesn't match the domain originaly set in the cookie, this might not clear what we want.
Still though, can't imagine if that is a real edge case or not.

Reply:
yah will do